automap/src/comm_layer/mod.rs

```diff
@@ -295,46 +293,46 @@ mod tests {
     #[test]
     fn should_crash_works() {
         vec![
-            (AutomapError::Unknown, true),
-            (AutomapError::NoLocalIpAddress, true),
+            (AutomapError::AllProtocolsFailed(vec![]), true),
             (AutomapError::CantFindDefaultGateway, true),
-            (AutomapError::IPv6Unsupported(Ipv6Addr::UNSPECIFIED), true),
+            (AutomapError::DeleteMappingError("".to_string()), false),
             (AutomapError::FindRouterError("".to_string()), true),
             (AutomapError::GetPublicIpError("".to_string()), true),
+            (AutomapError::HousekeeperAlreadyRunning, false),
+            (AutomapError::HousekeeperCrashed, false),
+            (AutomapError::IPv6Unsupported(Ipv6Addr::UNSPECIFIED), true),
+            (AutomapError::NoLocalIpAddress, true),
             (
-                AutomapError::SocketBindingError(
-                    "".to_string(),
-                    SocketAddr::from_str("0.0.0.0:0").unwrap(),
-                ),
+                AutomapError::PacketParseError(ParseError::WrongVersion(0)),
                 true,
             ),
+            (AutomapError::PermanentLeasesOnly, false),
+            (AutomapError::PermanentMappingError("".to_string()), true),
```

Comment:
Just a picking up a question, don't look the answer in my words. If you chose to attribute the `HousekeeperCrashed` error with `false`, then philosophically, what's the difference from an error like `PermanentMappingError(...)` or even a more interesting example. I noticed you have here these:
```
(AutomapError::PermanentMappingError("".to_string()), true),
...and...
(AutomapError::TemporaryMappingError("".to_string()), false),
```
I could attempt to interpret it as "if it's a temporary mapping, we might try again and again and we try the permanent one out only a single time" but that doesn't make sense so much. I'd be more open for a criticizing interpretation saying that `TemporaryMappingError` is very likely about to cause a cut-off of our Node since the last mapping has expired and therefore it has attributes for which might qualify as the opposite, with `true`. 
(Again, it's late here, and I'm having hard time to keep eyes open. Consider this more as food for thoughts than well formulated argumentation)

Reply:
It's a good catch that a `TemporaryMappingError` should stimulate more-often-than-normal retries, but A) that's not trivial, B) this card is already pretty big, and C) we don't have any idea yet how common `TemporaryMappingError`s are in real life. I'll make a card about that that we can play if they turn out to be a real problem.